boot/core/src/boot/task/built_in.clj

```diff
@@ -846,7 +846,8 @@
                 (boot.pom/pom-xml-parse-string ~(slurp pom))))
             pomname (when (and project version)
                       (str (name project) "-" version ".jar"))
-            fname   (or file pomname "project.jar")
+            file-parsed-string (string/replace file #"\{(\w+)\}" (fn [[_ group]] ((keyword group) {:project (name project) :version version})))
```

Comment:
```
(string/replace nil ,,,)
```
This will throw, no?

Reply:
The replacement function seems a bit hard to read and will throw if an unknown pattern is provided. 
Maybe something like the following could be used?

```clojure
(fn [[_ group]]
  (case group 
    "project" (name project)
    "version" version
    (do (util/warn "Unknown pattern {%s} in jar task's :file option\n" group)
        (str "{" group "}")))))
```
This will provide a helpful message if the user makes a typo instead of this:
```
NullPointerException   java.util.regex.Matcher.quoteReplacement (Matcher.java:701)
```

For readability the replacement function could also be put into it's own local var or even in a top-level `defn-`.